logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -14,60 +14,60 @@
+    This handler can be used to route Python standard logging messages directly
+    to the Stackdriver Logging API.
 
-from google.cloud.logging.handlers._helpers import format_stackdriver_json
+    This handler supports both an asynchronous and synchronous transport.
 
-_LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{pid}.json'
-_MAX_LOG_BYTES = 128 * 1024 * 1024
-_LOG_FILE_COUNT = 3
+    :type client: :class:`google.cloud.logging.client`
+    :param client: the authenticated Google Cloud Logging client for this
+                   handler to use
 
+    :type name: str
+    :param name: the name of the custom log in Stackdriver Logging. Defaults
+                 to 'python'. The name of the Python logger will be represented
+                 in the ``python_logger`` field.
 
-class AppEngineHandler(logging.handlers.RotatingFileHandler):
-    """A handler that writes to the App Engine fluentd Stackdriver log file.
+    :type transport: type
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth`.Transport.send`. Defaults to
+                      :class:`.BackgroundThreadTransport`. The other
+                      option is :class:`.SyncTransport`.
 
-    Writes to the file that the fluentd agent on App Engine Flexible is
-    configured to discover logs and send them to  Stackdriver Logging.
-    Log entries are wrapped in JSON and with appropriate metadata. The
-    process of converting the user's formatted logs into a JSON payload for
-    Stackdriver Logging consumption is implemented as part of the handler
-    itself, and not as a formatting step, so as not to interfere with
-    user-defined logging formats.
+    :type resource: :class:`~google.cloud.logging.resource.Resource`
+    :param resource: Monitored resource of the entry, defaults
+                     to the global resource type.
     """
 
-    def __init__(self):
-        """Construct the handler
-
-        Large log entries will get mangled if multiple workers write to the
-        same file simultaneously, so we'll use the worker's PID to pick a log
-        filename.
-        """
-        self.filename = _LOG_PATH_TEMPLATE.format(pid=os.getpid())
-        super(AppEngineHandler, self).__init__(self.filename,
-                                               maxBytes=_MAX_LOG_BYTES,
-                                               backupCount=_LOG_FILE_COUNT)
-
-    def format(self, record):
-        """Format the specified record into the expected JSON structure.
-
-        :type record: :class:`~logging.LogRecord`
-        :param record: the log record
-
-        :rtype: str
-        :returns: JSON str to be written to the log file
-        """
-        message = super(AppEngineHandler, self).format(record)
-        return format_stackdriver_json(record, message)
+    def __init__(self, client,
+                 name=EXCLUDED_LOGGER_DEFAULTS,
```

Comment:
This doesn't seem right? Also, shouldn't we be setting the name for them? (e.g., this isn't something the user should be able to configure).